src/parse_fc.c

```diff
@@ -243,22 +251,52 @@ struct policy_node *parse_fc_file(const char *filename, const struct string_list
 	struct policy_node *cur = head;
 
 	char *line = NULL;
+	char *ifdef_condition = NULL;
+	char *ifndef_condition = NULL;
+	char *token = NULL;
+	struct conditional_data *conditional = NULL;
 
 	ssize_t len_read = 0;
 	size_t buf_len = 0;
 	unsigned int lineno = 0;
+	bool is_within_ifdef = false;
+	bool is_within_ifndef = false;
 	while ((len_read = getline(&line, &buf_len, fd)) != -1) {
 		lineno++;
 		if (len_read <= 1 || line[0] == '#') {
 			continue;
-		}
-		// Skip over m4 constructs
-		if (strncmp(line, "ifdef", 5) == 0 ||
-		    strncmp(line, "ifndef", 6) == 0 ||
-		    strncmp(line, "')", 2) == 0 ||
-		    strncmp(line, "', `", 4) == 0 ||
-		    strncmp(line, "',`", 3) == 0) {
-
+		} else if (!strncmp(line, "ifdef", 5)) {
+			is_within_ifdef = true;
+			token = strtok(line, "`");
+			if (token) {
+				token = strtok(NULL, "`");
+				token = strtok(token, "'");
```

Comment:
Shouldn't this be `strtok(NULL, "'");"` since you're continuing in the same string?

Reply:
yes that is a mistake.